### PR TITLE
VXFM-7669 - RDM Drive replacement operation

### PR DESCRIPTION
**Pull request by krp2**
@saxens7 @gavin-scott @sushilrai   Can you please review and merge?

### CODE REVIEW

**Comment by saxens7** [comment]
https://github.com/dell-asm/asm-deployer/pull/3007
https://github.com/dell-asm/asm-core/pull/5242

File: lib/puppet/provider/vc_vm_rdm/default.rb [hunk 1]
```diff
@@ -41,6 +41,15 @@ def remove_rdm_disks
   end
 
   def add_rdm_disks
+    rdm_disk_details = resource[:rdm_disk_details]
+    disk_id_to_add = rdm_disk_details.collect do |_, disk_facts|
+      disk_facts["OtherUIDs"]
+    end
+    rdm_disks = vm.config.hardware.device.find_all do |device|
+      device if device.class == RbVmomi::VIM::VirtualDisk && device.backing.is_a?(RbVmomi::VIM::VirtualDiskRawDiskMappingVer1BackingInfo) &&
```
Comment (sushilrai) [inline lib/puppet/provider/vc_vm_rdm/default.rb]: indent all conditions. Keep all conditions in new line

File: lib/puppet/provider/vc_vm_rdm/default.rb [hunk 2]
```diff
@@ -41,6 +41,15 @@ def remove_rdm_disks
...
+          disk_id_to_add.include?(device.backing.deviceName)
+    end
+    return unless rdm_disks.empty?
```
Comment (sushilrai) [inline lib/puppet/provider/vc_vm_rdm/default.rb]: add a newline after return statement

Comment (krp2) [reply lib/puppet/provider/vc_vm_rdm/default.rb]: done

File: lib/puppet/provider/vc_vm_rdm/default.rb (hunk 1 above)
Comment (krp2) [reply lib/puppet/provider/vc_vm_rdm/default.rb]: done

**Pull request merged**